scripts/emergency-rollback.sh

```
@@ -50,6 +62,75 @@ init_logging() {
+            echo "[$(date +'%Y-%m-%d %H:%M:%S')] Vercel authenticated" | tee -a "$LOG_FILE"
+        fi
+    else
+        echo "[$(date +'%Y-%m-%d %H:%M:%S')] Using Vercel token authentication" | tee -a "$LOG_FILE"
+    fi
+    
+    # Check Node.js
+    if ! command -v node >/dev/null 2>&1; then
+        echo "[$(date +'%Y-%m-%d %H:%M:%S')] ERROR: Node.js not found" | tee -a "$LOG_FILE"
+        ((validation_errors++))
+    else
+        local node_version=$(node --version | sed 's/v//' | cut -d'.' -f1)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)